common/rpc/encryption/localStorePerHostCertProviderMap.go

```diff
@@ -46,28 +47,36 @@ func newLocalStorePerHostCertProviderMap(overrides map[string]config.ServerTLS)
 	}
 
 	factory.certProviderCache = make(map[string]*localStoreCertProvider, len(overrides))
+	factory.clientAuthCache = make(map[string]bool, len(overrides))
 
 	for host, settings := range overrides {
-		factory.certProviderCache[strings.ToLower(host)] = &localStoreCertProvider{
+		lcHost := strings.ToLower(host)
+		factory.certProviderCache[lcHost] = &localStoreCertProvider{
 			tlsSettings: &config.GroupTLS{
 				Server: settings,
 			},
 		}
+		factory.clientAuthCache[lcHost] = settings.RequireClientAuth
 	}
 
 	return factory
 }
 
-func (f *localStorePerHostCertProviderMap) GetCertProvider(hostName string) (CertProvider, error) {
+func (f *localStorePerHostCertProviderMap) GetCertProvider(hostName string,
+) (provider CertProvider, clientAuthRequired bool, err error) {
```

Comment:
[Nit] no point naming the return variables if you are just going to return values at each point and if you are not using any of them in a defer statement